server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -1142,6 +1142,7 @@ public void perform(Request request, ActionListener<PrimaryResult<ReplicaRequest
                 listener = listener.map(result -> {
                     assert result.replicaRequest() == null || result.finalFailure == null
                         : "a replica request [" + result.replicaRequest() + "] with a primary failure [" + result.finalFailure + "]";
+                    assert result.replicaRequest().getParentTask() != null : "a replica request must have a parent task id";
```

Comment:
Unfortunately this assertion is now vacuous, the parent task ID is `EMPTY_TASK_ID` if unset, never `null`.

It looks like we don't set the parent task properly in a few places, and IMO that's a bug. We can back this out for now and follow-up with fixes for this.